modular_skyrat/modules/conveyor_sorter/code/conveyor_sorter.dm

```diff
@@ -145,3 +143,59 @@
 		"conveysorter",
 	)
 	research_costs = list(TECHWEB_POINT_TYPE_GENERIC = 2500)
+
+/obj/item/conveyor_sorter/improved
+	name = "improved conveyor sorter lister"
+	desc = "A tool that is used to not only create the conveyor sorters, but give lists to the conveyor sorters. This one has an increased capacity of eight."
+	icon_state = "lister_improved"
+	max_sorters = 8
+
+/obj/item/conveyor_sorter/improved/examine(mob/user)
+	. = ..()
+	. += span_notice("This version of the sorter allows up to eight sorters to be placed.")
+
+/obj/item/conveyor_sorter/improved/attack_self(mob/user, modifiers)
+	if(length(spawned_sorters) >= max_sorters)
+		to_chat(user, span_warning("You may only have [max_sorters] spawned conveyor sorters!"))
+		return
+	var/obj/effect/decal/cleanable/conveyor_sorter/improved/new_cs = new /obj/effect/decal/cleanable/conveyor_sorter/improved(get_turf(src))
+	new_cs.parent_item = src
+	new_cs.sorting_list = src.current_sort
+	spawned_sorters += new_cs
+
+/obj/effect/decal/cleanable/conveyor_sorter/improved
+	name = "improved conveyor sorter"
+	desc = "A mark that will sort items out based on what they are. This one can sort in multiple directions."
+	icon = 'modular_skyrat/modules/conveyor_sorter/icons/conveyor_sorter.dmi'
+	icon_state = "sorter_improved"
+	light_range = 3
+	light_color = COLOR_BLUE_LIGHT
+
+/obj/effect/decal/cleanable/conveyor_sorter/improved/attack_hand(mob/living/user, list/modifiers)
+	var/user_choice_improved = tgui_input_list(user, "Choose which direction to sort to!", "Direction choice", list("North", "East", "South", "West", "NorthEast", "NorthWest", "SouthEast", "SouthWest"))
+	if(!user_choice_improved)
+		return ..()
+	var/dir = text2dir(user_choice_improved)
+
+	setDir(dir)
+
+	visible_message("[src] pings, updating its sorting direction!")
+	playsound(src, 'sound/machines/ping.ogg', 30, TRUE)
+
```

Comment:
This proc is nearly exactly the same as the parent proc, and therefore completely useless.

Just make the list of directions that can be used a variable on the `/obj/effect/decal/cleanable/conveyor_sorter` instead, and use that list in the proc of the parent.
```suggestion
```